app/views/repositories/annotate.html.erb

```diff
@@ -34,25 +34,40 @@ See doc/COPYRIGHT.rdoc for more details.
              locals: { path: @path, revision: @rev }.merge(kind: 'file') %>
 </div>
 <p><%= render partial: 'link_to_functions' %></p>
-<% colors = Hash.new {|k,v| k[v] = (k.size % 12) } %>
-<div class="autoscroll">
-  <table class="filecontent annotate CodeRay">
-    <tbody>
-      <% line_num = 1 %>
-      <% syntax_highlight(@path, to_utf8_for_repositories(@annotate.content)) do |line| %>
-        <% revision = @annotate.revisions[line_num-1] %>
-        <tr class="bloc-<%= revision.nil? ? 0 : colors[revision.identifier || revision.revision] %>">
-          <th class="line-num" id="L<%= line_num %>"><a href="#L<%= line_num %>"><%= line_num %></a></th>
-          <td class="revision">
-            <%= (revision.identifier ? link_to_revision(revision, @project) : format_revision(revision)) if revision %></td>
-          <td class="author"><%= h(revision.author.to_s.split('<').first) if revision %></td>
-          <td class="line-code">
-            <pre><%= line %></pre>
-          </td>
-        </tr>
-        <% line_num += 1 %>
-      <% end %>
-    </tbody>
-  </table>
-</div>
-<% html_title(l(:button_annotate)) -%>
+
+<% if @annotate.nil? || @annotate.empty? %>
+  <div class="generic-table--container">
+    <div class="generic-table--no-results-container">
+      <h2 class="generic-table--no-results-title">
+        <i class="icon-info"></i>
+        <%= l(:label_nothing_display) %>
+      </h2>
+      <div class="generic-table--no-results-description">
+        <p class="nodata"><%= l('repositories.warnings.cannot_annotate') %></p>
+      </div>
+    </div>
+  </div>
+<% else %>
+  <% colors = Hash.new {|k,v| k[v] = (k.size % 12) } %>
+  <div class="autoscroll">
+    <table class="filecontent annotate CodeRay">
+      <tbody>
+        <% line_num = 1 %>
+        <% syntax_highlight(@path, to_utf8_for_repositories(@annotate.content)) do |line| %>
+          <% revision = @annotate.revisions[line_num-1] %>
+          <tr class="bloc-<%= revision.nil? ? 0 : colors[revision.identifier || revision.revision] %>">
+            <th class="line-num" id="L<%= line_num %>"><a href="#L<%= line_num %>"><%= line_num %></a></th>
+            <td class="revision">
+              <%= (revision.identifier ? link_to_revision(revision, @project) : format_revision(revision)) if revision %></td>
+            <td class="author"><%= h(revision.author.to_s.split('<').first) if revision %></td>
+            <td class="line-code">
+              <pre><%= line %></pre>
+            </td>
+          </tr>
+          <% line_num += 1 %>
+        <% end %>
+      </tbody>
+    </table>
+  </div>
+  <% html_title(l(:button_annotate)) -%>
```

Comment:
You should always add an html title, not just in the `else` case.

I propose moving this up, so that it is not so easy to miss.

Also: WTF `button_annotate`? Button? ^^